### PR TITLE
Update nano and fix coreutils on Darwin

### PR DESCRIPTION
**Pull request by DiegoMagdaleno**
**Coreutils:**
-  Fixes an error making it not being able to build if you are root user
- Fixed errors building on Darwin, [reference](https://github.com/Homebrew/homebrew-core/blob/master/Formula/coreutils.rb)
- Add the program prefix "g" so we don't conflict with Apple's "coreutils" that are BSD versions

**Nano:**
- Updated nano version up to 4.9, but also included previous versions

### CODE REVIEW

**Comment by DiegoMagdaleno** [comment]
I don't know why the test for the documentation failed, if I didn't modify any of those :/

**Comment by adamjstewart** [comment]
> I don't know why the test for the documentation failed, if I didn't modify any of those :/

This morning Sphinx released 3.0.0, which added a lot of fancy features like automatic type checking. Unfortunately/fortunately, these new features uncovered a bug in our Executable class. This has been fixed in develop, so if you rebase your PR on the tip of develop, it should pass the doc tests.

File: var/spack/repos/builtin/packages/coreutils/package.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# -*- coding: utf-8 -*-
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/coreutils/package.py:1]: This change probably isn't necessary.

File: var/spack/repos/builtin/packages/coreutils/package.py [hunk 2]
```diff
@@ -7,13 +8,15 @@
 
 
 class Coreutils(AutotoolsPackage, GNUMirrorPackage):
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/coreutils/package.py:11]: This change probably isn't necessary

File: var/spack/repos/builtin/packages/coreutils/package.py [hunk 3]
```diff
@@ -22,3 +25,14 @@ class Coreutils(AutotoolsPackage, GNUMirrorPackage):
     version('8.23', sha256='ec43ca5bcfc62242accb46b7f121f6b684ee21ecd7d075059bf650ff9e37b82d')
 
     build_directory = 'spack-build'
+
+    def configure_args(self):
+        spec = self.spec
+        configure_args = []
+        if spec.satisfies('platform=darwin'):
+            configure_args.append('--program-prefix=g')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/coreutils/package.py:33]: I _think_ I'm okay with this decision, but let's wait and here from other users.

File: var/spack/repos/builtin/packages/coreutils/package.py (hunk 1 above)
Comment (DiegoMagdaleno) [reply var/spack/repos/builtin/packages/coreutils/package.py:1]: Okay I make sure to remove unnecessary chances and sure hope the decision of using the "g" prefix gets approved, I will make sure to commit the unnecessary changes removed.

**Comment by DiegoMagdaleno** [comment]
Unnecessary changes are removed :)

**Comment by rsumner868** [comment]
Looks good or so i Think

File: var/spack/repos/builtin/packages/libgit2/package.py [hunk 4]
```diff
@@ -96,3 +97,22 @@ def cmake_args(self):
...
+    @run_after('install')
+    def fix_linking(self):
+        if sys.platform == 'darwin':
+            install_name_tool = which('install_name_tool')
+            install_name_tool('-change', 'libssh2.1.dylib',
+                              '{0}'.format(
+                                  self.spec['libssh2'].prefix.lib
+                                  + '/libssh2.1.0.1.dylib'),
+                              '{0}'.format(
+                                  prefix.lib + '/libgit2.1.0.0.dylib'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libgit2/package.py:118]: Does this work?
```python
    @run_after('install')                                                       
    def darwin_fix(self):                                                       
        # The shared library is not installed correctly on Darwin; fix this     
        if self.spec.satisfies('platform=darwin'):                              
            fix_darwin_install_name(self.prefix.lib)       
```
We have a builtin function to do this.

**Comment by adamjstewart** [comment]
It seems like these changes are unrelated. It's best to open a separate PR for each package change, as it makes it easier to review.

**Comment by DiegoMagdaleno** [comment]
Oh yeah my bad, didn’t realise I committed this changes to the develop branch on my fork, I work on my own branch for that changes, will revert, my bad.

**Comment by DiegoMagdaleno** [comment]
Just reverted, sorry will work on my own branch from now on, forgot to do it, the only changes here are Nano and the coreutils, however, I will do a pull request for each package I fix or add, again sorry, and yeah I try the method you suggested in the code review for fixing libgit.

File: var/spack/repos/builtin/packages/coreutils/package.py (hunk 3 above)
Comment (tgamblin) [reply var/spack/repos/builtin/packages/coreutils/package.py:31]: Ok I'll bite.  I use this without the prefix all the time.  I think this should be an option, like `+prefix`.  Homebrew ships coreutils with `libexec/gnubin` containing the non-prefixed binaries, so you can actually add it to your path.

Was there an issue with using this without the prefix?  I've been using it that way on my mac for a while.

**Pull request merged**